addons/auth_signup/models/base_config_settings.py

```diff
@@ -29,7 +32,7 @@ def set_values(self):
         set_param = self.env['ir.config_parameter'].sudo().set_param
         # we store the repr of the values, since the value of the parameter is a required string
         set_param('auth_signup.reset_password', repr(self.auth_signup_reset_password))
-        set_param('auth_signup.allow_uninvited', repr(self.auth_signup_uninvited))
+        set_param('auth_signup.allow_uninvited', repr(self.auth_signup_uninvited == 'b2c'))
```

Comment:
Well I choose not to change it to avoid having to migrate, update code using it and break community addons that might use it. Moreover changing the stored value would imply to change the parameter name (now allow_uninvited but invitation_type or something like that). So I prefer to keep it like this currently.